### PR TITLE
XW-754 fix bugs found while testing ES6

### PR DESCRIPTION
**Pull request by bognix**
E.g.: fix calling track inside track and stop using arguments in Ooyala

### CODE REVIEW

**Comment by vforge** [comment]
LGTM

File: front/scripts/mercury/modules/VideoPlayers/Ooyala.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ export default class OoyalaPlayer extends BasePlayer {
 	 */
 	setupPlayer() {
 		this.params = $.extend(this.params, {
-			onCreate: () => {
-				return this.onCreate.apply(this, arguments);
+			onCreate: (...args) => {
```
Comment (hakubo) [inline front/scripts/mercury/modules/VideoPlayers/Ooyala.js]: yummy!

**Comment by hakubo** [comment]
LGTM

File: front/scripts/mercury/modules/VideoPlayers/Ooyala.js (hunk 1 above)
Comment (rogatty) [inline front/scripts/mercury/modules/VideoPlayers/Ooyala.js]: Nice catch, citing http://blog.getify.com/arrow-this/:
>In fact, => arrow functions do not bind a this, arguments, super (ES6), or new.target (ES6).

File: front/scripts/mercury/modules/VideoPlayers/Base.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import load from '../../utils/load';
 import {containerSize} from '../../utils/calculation';
-import {track} from '../../utils/track';
+import {track as utilsTrack} from '../../utils/track';
```
Comment (rogatty) [inline front/scripts/mercury/modules/VideoPlayers/Base.js]: In other places we use `as mercuryTrack`. I'm not saying it's better but would be nice to have it unified either way.

Comment (bognix) [inline front/scripts/mercury/modules/VideoPlayers/Base.js]: makes sense :+1:

**Comment by hakubo** [comment]
lgtm, branches are cheap, try to make new ones for additional stuff

File: front/scripts/mercury/modules/VideoPlayers/Base.js (hunk 2 above)
Comment (rogatty) [inline front/scripts/mercury/modules/VideoPlayers/Base.js]: @bognix so what do we do with it? :)

**Comment by bognix** [comment]
@hakubo it was easier for me to do it on the same branch so I don't have to switch context every time I make a change. So far changes are so small that keeping them in one PR and branch shouldn't decrease readability.

File: front/scripts/mercury/modules/VideoPlayers/Base.js (hunk 2 above)
Comment (bognix) [inline front/scripts/mercury/modules/VideoPlayers/Base.js]: I'm gonna do it. Work is not finished yet :) Don't you worry :)

File: server/facets/operations/prepareArticleData.ts [hunk 3]
```diff
@@ -58,8 +60,6 @@ function prepareArticleData(request: Hapi.Request, data: ArticlePageData): any {
 
 	result.displayTitle = title;
 	result.htmlTitle = (htmlTitle) ? htmlTitle : Utils.getHtmlTitle(wikiVariables, title);
-	result.isMainPage = articleData.isMainPage;
-	result.themeColor = Utils.getVerticalColor(localSettings, wikiVariables.vertical);
```
Comment (rogatty) [inline server/facets/operations/prepareArticleData.ts]: Was it removed on purpose?

Comment (bognix) [inline server/facets/operations/prepareArticleData.ts]: second line is a mistake! nice catch!

**Comment by hakubo** [comment]
LGTM

**Comment by rogatty** [comment]
LGTM

**Pull request merged**